lstchain/tools/lstchain_create_drs4_pedestal_file.py

```diff
@@ -0,0 +1,109 @@
+"""
+Create drs4 time pedestal fits file.
+"""
+import numpy as np
+from astropy.io import fits
+
+from ctapipe.core import Provenance, traits
+from ctapipe.core import Tool
+from ctapipe.io import EventSource
+from lstchain.calib.camera.r0 import LSTR0Corrections
+from lstchain.calib.camera.drs4 import DragonPedestal
+
+
+class PedestalFITSWriter(Tool):
+
+    name = "PedestalFITSWriter"
+    description = "Generate a pedestal FITS file"
+
+    output_file = traits.Path(
+        default_value="pedestal.fits",
+        help="Path to the generated FITS pedestal file",
+        directory_ok=False,
+        exists=False,
+    ).tag(config=True)
+
+    deltaT = traits.Bool(
+        help="Flag to use deltaT correction. Default=True", default_value=True
+    ).tag(config=True)
+
+    aliases = {
+        "input_file": "EventSource.input_url",
+        "output_file": "PedestalFITSWriter.output_file",
+    }
+
+    flags = {
+        "deltaT": ({"PedestalFITSWriter": {"deltaT": True}}, "Flag to use deltaT correction. Default is True")
+    }
+
+    classes = [EventSource, LSTR0Corrections, DragonPedestal]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a pedestal FITS file for low level calibration.
+
+        For getting help run:
+        lstchain_create_drs4_pedestal_file --help
+        """
+
+        self.eventsource = None
+        self.pixel_ids = None
+        self.pedestal = None
+        self.lst_r0 = None
+
+    def setup(self):
+
+        self.log.debug("Opening file")
+        self.eventsource = EventSource.from_config(parent=self)
```

Comment:
Also call `add_component` here, the eventsource also has config option.

Reply:
If I replace  `EventSource.from_config(parent=self)` by `self.add_component(EventSource(parent=self))` or by `EventSource(parent=self)` I get the following error:

```
ERROR [PedestalFITSWriter] (tool/run): Caught unexpected exception: Can't instantiate abstract class EventSource with abstract methods _generator, datalevels, is_compatible, is_simulation, obs_id, subarray
Traceback (most recent call last):
  File "/opt/miniconda3/envs/lst-dev/lib/python3.7/site-packages/ctapipe/core/tool.py", line 215, in run
    self.setup()
  File "/Users/jer/git/cta-observatory/cta-lstchain/lstchain/tools/lstchain_create_drs4_pedestal_file.py", line 65, in setup
    self.eventsource = self.add_component(EventSource(parent=self))
  File "/opt/miniconda3/envs/lst-dev/lib/python3.7/site-packages/traitlets/traitlets.py", line 955, in __new__
    inst = new_meth(cls)
TypeError: Can't instantiate abstract class EventSource with abstract methods _generator, datalevels, is_compatible, is_simulation, obs_id, subarray
```

Reply:
Ok that's probably a bug, but `add_component` does not exist in ctapipe 0.10 anymore anyways, so just leave it out here I'd say.